osdep/win32-console-wrapper.c

```diff
@@ -38,26 +38,31 @@ static void cr_perror(const wchar_t *prefix)
 
 static int cr_runproc(wchar_t *name, wchar_t *cmdline)
 {
-    STARTUPINFOW si;
-    STARTUPINFOW our_si;
-    PROCESS_INFORMATION pi;
     DWORD retval = 1;
 
-    ZeroMemory(&si, sizeof(si));
-    si.cb = sizeof(si);
-    si.hStdInput = GetStdHandle(STD_INPUT_HANDLE);
-    si.hStdOutput = GetStdHandle(STD_OUTPUT_HANDLE);
-    si.hStdError = GetStdHandle(STD_ERROR_HANDLE);
-    si.dwFlags |= STARTF_USESTDHANDLES;
-
     // Copy the list of inherited CRT file descriptors to the new process
-    our_si.cb = sizeof(our_si);
+    STARTUPINFOW our_si = {sizeof(our_si)};
```

Comment:
Why setting `cb` here? `GetStartupInfo` only sends values to the structure, and any existing value will get overwritten.

Reply:
`cb` is the size of structure. It is a way of versioning. `GetStartupInfoW` could check this field and decide what fields should be filled. This is common in win32 api, getters require this one field to be set.

Reply:
It is just how it works, maybe in this particular case, they just overwrite whole struct with own version, but in principle size args should always be set by the caller. 

See for example:
https://learn.microsoft.com/en-us/windows/win32/api/dbghelp/nf-dbghelp-symgetlinefromaddr64
or 
https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-systemparametersinfow with GET param
they would fail if size is not set correctly.

And similarly here if STARTUPINFO ever get extended the size may be checked.